tempodb/tempodb.go

```diff
@@ -209,8 +209,8 @@ func (rw *readerWriter) CompleteBlock(block *wal.AppendBlock, combiner model.Obj
 // new block will have the same ID as the input block.
 func (rw *readerWriter) CompleteBlockWithBackend(ctx context.Context, block *wal.AppendBlock, combiner model.ObjectCombiner, r backend.Reader, w backend.Writer) (common.BackendBlock, error) {
 
-	// Try to use same version and encoding as the WAL
-	vers, err := encoding.FromVersion(block.Meta().Version)
+	// The destination block format:
+	vers, err := encoding.FromVersion(rw.cfg.Block.Version)
```

Comment:
do we test this value to make sure it's valid before this line? otherwise an ingester will start up find and then only begin failing later when it starts trying to complete blocks.

Reply:
Yep, it was added to `config.go/validateConfig()`.